test/test_backend.py

```diff
@@ -170,7 +171,99 @@ def test_counts_over_time(self):
                     # Part 2: all other row counts should be non-zero
                     self.assertGreater(df[col][row], 0, msg=f"Table '{row}' had 0 rows in run '{col}'")
 
-
+    def test_subgraph(self):
```

Comment:
As long as it runs and passes, looks good to me! Take a lot of space vertically like this but that is probably the most standard way to represent it in terms of code style.